observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -0,0 +1,189 @@
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final StringListAttributeDefinition EXPOSED_SUBSYSTEMS =
+            new StringListAttributeDefinition.Builder("exposed-subsystems")
+                    .setDefaultValue(ModelNode.fromJSONString("[\"*\"]"))
+                    .setRequired(false)
+                    .setRestartAllServices()
+                    .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = {
+            EXPOSED_SUBSYSTEMS,
+            ENDPOINT,
+            STEP
+    };
+
+    private WildFlyRegistry wildFlyRegistry;
+
+    @Override
+    public ManagementResourceRegistration register(SubsystemRegistration parent,
+                                                   ManagementResourceRegistrationContext context) {
+        ManagementResourceRegistration registration =
+                parent.registerSubsystemModel(ResourceDefinition.builder(ResourceRegistration.of(PATH), RESOLVER).build());
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(RESOLVER)
+                .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+                .addCapability(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY)
+                .addCapability(MICROMETER_REGISTRY_RUNTIME_CAPABILITY)
+                .addAttributes(List.of(ENDPOINT, STEP, EXPOSED_SUBSYSTEMS))
+                .build();
+
+        ManagementResourceRegistrar.of(descriptor).register(registration);
+
+        return registration;
+    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        List<String> exposedSubsystems = MicrometerSubsystemRegistrar.EXPOSED_SUBSYSTEMS.unwrap(context, model);
+        boolean exposeAnySubsystem = exposedSubsystems.remove("*");
+        String endpoint = MicrometerSubsystemRegistrar.ENDPOINT.resolveModelAttribute(context, model)
+                .asStringOrNull();
+        Long step = MicrometerSubsystemRegistrar.STEP.resolveModelAttribute(context, model)
+                .asLong();
+
+        WildFlyMicrometerConfig config = new WildFlyMicrometerConfig(endpoint, step);
+        wildFlyRegistry = new WildFlyOtlpRegistry(config);
+
+        context.addStep(new AbstractDeploymentChainStep() {
+            @Override
+            public void execute(DeploymentProcessorTarget processorTarget) {
+                processorTarget.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, DEPENDENCIES,
+                        DEPENDENCIES_MICROMETER, new MicrometerDependencyProcessor());
+                processorTarget.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, POST_MODULE,
+                        POST_MODULE_MICROMETER,
+                        new MicrometerDeploymentProcessor(exposeAnySubsystem, exposedSubsystems, wildFlyRegistry));
+            }
+        }, RUNTIME);
```

Comment:
Move this to your ResourceDescriptor, e.g.
```java
.withDeploymentChainContributor(new Consumer<>() {
    @Override
    public void accept(DeploymentProcessorTarget target) {
        //...
    }
}
```

Reply:
I had it that way, but I need model values. How/where would I cache those for use here?

Reply:
I think I've found something works. Thoughts welcome. :)